File: server/Models/Property.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.Metrics;

namespace RedHouse_Server.Models
{
    public class Property
    {
        [Key]
        public int Id { get; set; }
        public string PropertyCode { get; set; }

        public string PropertyType { get; set; }
        [ForeignKey(nameof(User))]
        public int UserId { get; set; }
        [ForeignKey(nameof(Location))]
        public int LocationId { get; set; }
        public Location Location { get; set; }

        // [ForeignKey(nameof(Neighborhood))]
        // public int NeighborhoodId { get; set; }
        public int Price { get; set; }
        public int NumberOfBedRooms { get; set; }
        public int NumberOfBathRooms { get; set; }
        public float SquareMetersArea { get; set; }
        public string PropertyDescription { get; set; }
        public DateTime BuiltYear { get; set; } = DateTime.Now;
        public string View { get; set; }
        public DateTime AvailableOn { get; set; } = DateTime.Now;
        public string PropertyStatus { get; set; } = "Accepting offers";
        public int? NumberOfUnits { get; set; } = 1;
        public int ParkingSpots { get; set; }
        public string ListingType { get; set; }
        public string IsAvailableBasement { get; set; } = "false";
        public string ListingBy { get; set; } = "Landlord";
        public ICollection<PropertyFile>? propertyFiles { get; set; }
    }
}

File: client/lib/model/property.dart
import 'package:client/model/location.dart';
import 'package:client/model/property_files.dart';
import 'package:client/model/user.dart';
import 'package:intl/intl.dart';
import 'package:json_annotation/json_annotation.dart';
part "property.g.dart";

@JsonSerializable()
class Property {
  int id;
  String propertyType;
  int userId;
  User? user;
  String propertyCode;
  int locationId;
  List<PropertyFile>? propertyFiles;
  Location? location;
  int price;
  int numberOfBedRooms;
  int numberOfBathRooms;
  int squareMetersArea;
  String propertyDescription;
  DateTime builtYear;
  String view;
  DateTime availableOn;
  String propertyStatus;
  int numberOfUnits;
  int parkingSpots;
  String listingType;
  String isAvailableBasement;
  String listingBy;

  Property({
    required this.id,
    required this.propertyType,
    required this.userId,
    this.user,
    required this.propertyCode,
    this.propertyFiles,
    required this.locationId,
    this.location,
    required this.price,
    required this.numberOfBedRooms,
    required this.numberOfBathRooms,
    required this.squareMetersArea,
    required this.propertyDescription,
    required this.builtYear,
    required this.view,
    required this.availableOn,
    required this.propertyStatus,
    required this.numberOfUnits,
    required this.parkingSpots,
    required this.listingType,
    required this.isAvailableBasement,
    required this.listingBy,
  });

  factory Property.fromJson(Map<String, dynamic> json) =>
      _$PropertyFromJson(json);
  Map<String, dynamic> toJson() => _$PropertyToJson(this);

}
